clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java

```diff
@@ -73,5 +80,6 @@ public void onReceive(String message) throws Exception {
     void onClose(int code, String reason) {
         logger.log(LogLevel.Information, "WebSocket connection stopping with " +
                 "code %d and reason '%s'.", code, reason);
+        onClose.accept(reason);
```

Comment:
We should only pass a value here if `code != 1000`

Reply:
It looks like if the WebSocket `onFailure` method is called then `onClosing` wont be called, so we might need to invoke the close callback on that path too.

Reply:
We should test that scenario